src/CommonDBTM.php

```diff
@@ -887,7 +887,7 @@ public function cleanRelationData()
                 foreach ($fields as $field) {
                     if (is_array($field)) {
                         // Relation based on 'itemtype'/'items_id' (polymorphic relationship)
-                        if ($this instanceof IPAddress && in_array('mainitemtype', $field) && in_array('mainitems_id', $field)) {
+                        if (($this instanceof IPAddress || $tablename == IPAddress::getTable()) && in_array('mainitemtype', $field) && in_array('mainitems_id', $field)) {
```

Comment:
Check was not done on the correct item.

```suggestion
                        if ($itemtype instanceof IPAddress && in_array('mainitemtype', $field) && in_array('mainitems_id', $field)) {
```